qa/rpc-tests/mergetoaddress_helper.py

```diff
@@ -15,14 +15,18 @@
 
 
 def assert_mergetoaddress_exception(expected_error_msg, merge_to_address_lambda):
+    caught_exception = True
     try:
         merge_to_address_lambda()
-        fail("Expected exception: %s" % expected_error_msg)
+        caught_exception = False
     except JSONRPCException as e:
         assert_equal(expected_error_msg, e.error['message'])
     except Exception as e:
         fail("Expected JSONRPCException. Found %s" % repr(e))
 
+    if not caught_exception:
+        fail("Expected exception: %s" % expected_error_msg)
```

Comment:
This can be written more simply as:
```
    try:
        merge_to_address_lambda()
    except JSONRPCException as e:
        assert_equal(expected_error_msg, e.error['message'])
    except Exception as e:
        fail("Expected JSONRPCException. Found %s" % repr(e))
    else:
        fail("Expected exception: %s" % expected_error_msg)
```

(The `else:` clause executes if the `try:` clause completes normally, but is not itself within the scope of the `try`.)